### PR TITLE
Optimize scheduler host-side batching overhead

### PR DESCRIPTION
**Pull request by AmedeoBiolatti**
## Motivation
Reduce host-side scheduler overhead in the prefill path, especially in `Scheduler.get_next_batch_to_run()`.

This addresses the scheduler overhead discussed in https://github.com/sgl-project/sglang-jax/issues/293.

The goal is behavior parity with lower Python/Numpy overhead during request admission and batch preparation.

## Modifications
- Avoid unnecessary `fill_ids` copies when a request has no generated output yet.
- Skip radix-prefix matching work when the prefix cache is disabled.
- Replace Python list flattening via `sum(..., [])` with `np.fromiter`.
- Avoid rebuilding `set(all_can_run_reqs)` for every waiting-queue element.
- Use a narrower post-admission budget check in `PrefillAdder`.
- Batch several logits metadata host-to-device transfers through one `device_array` call.
- Fill extend positions directly into the target numpy array.

## Benchmarking and Profiling
I created a small micro-benchmark for `Scheduler.get_next_batch_to_run()` in isolation. I link the gist: https://gist.github.com/AmedeoBiolatti/a4d04e53b3e0640fb1a98b00ae8056eb

Command used locally:
`python bench_get_next_batch_to_run.py --iterations 300 --warmup 30 --queue-size 512 --dp-size 4 --max-running-requests 512`

Results:
| main | this |
|--|--|
| 1.776 ms | 1.122 ms |

Improvement: ~36.8%.

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/managers/schedule_policy.py
Comment (cjx0709) [inline python/sgl_jax/srt/managers/schedule_policy.py:396]: if not one use this function, you can delete it.(And maybe update the doc:"docs/architecture/03-scheduler.md"

Comment (AmedeoBiolatti) [reply python/sgl_jax/srt/managers/schedule_policy.py:396]: I thinks this is still used in one place

(docs updated!)

**Review by cjx0709 — APPROVED** [review]
LGTM, will be merged when ci pass.
